impl/src/main/java/org/ehcache/spi/loaderwriter/DefaultCacheLoaderWriterFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.loaderwriter;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.loaderwriter.DefaultCacheLoaderWriterConfiguration;
+import org.ehcache.config.loaderwriter.DefaultCacheLoaderWriterFactoryConfiguration;
+import org.ehcache.internal.classes.ClassInstanceProvider;
+
+/**
+ * @author Alex Snaps
+ */
+public class DefaultCacheLoaderWriterFactory extends ClassInstanceProvider<CacheLoaderWriter<?, ?>> implements CacheLoaderWriterFactory {
+
+  public DefaultCacheLoaderWriterFactory() {
+    super(DefaultCacheLoaderWriterFactoryConfiguration.class, DefaultCacheLoaderWriterConfiguration.class);
+  }
+
+  @Override
+  public <K, V> CacheLoaderWriter<? super K, V> createCacheLoaderWriter(final String alias, final CacheConfiguration<K, V> cacheConfiguration) {
+    return (CacheLoaderWriter<? super K, V>) newInstance(alias, cacheConfiguration);
```

Comment:
I lied, one more comment. This is an unchecked cast. 

I can't think of how this thing could become aware of the <K, V> it is making instances for. 

Suppress the warning?